src/App.tsx

```diff
@@ -15,47 +15,95 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+type SortType = 'NONE' | 'ALPHABET' | 'LENGTH';
+type IsReversed = boolean;
+type IsReset = boolean;
+
 export const App: React.FC = () => {
+  const [goods, setGoods] = useState([...goodsFromServer]);
+  const [sortType, setSortType] = useState<SortType>('NONE');
+  const [isReversed, setReverse] = useState<IsReversed>(false);
+  const [isReset, setReset] = useState<IsReset>(false);
```

Comment:
TypeScript can infer the types of the state variables isReversed and isReset based on the initial values false. Since the initial values are booleans, TypeScript will infer the types as boolean automatically.
```suggestion
  const [isReversed, setReverse] = useState(false);
  const [isReset, setReset] = useState(false);
```